xarray/testing/assertions.py

```diff
@@ -95,6 +95,18 @@ def assert_isomorphic(a: DataTree, b: DataTree, from_root: bool = False):
         raise TypeError(f"{type(a)} not of type DataTree")
 
 
+def align_dims(a, b, check_dims):
```

Comment:
```suggestion
def maybe_transpose_dims(a, b, check_dims):
```

Factoring this out into a function is neat, but "align" usually means something else in xarray, so let's call this function something else instead.